File: chrome/test/data/nacl/simple.cc
// Copyright 2012 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <string>

#include "ppapi/cpp/instance.h"
#include "ppapi/cpp/module.h"
#include "ppapi/cpp/var.h"

class SimpleInstance : public pp::Instance {
 public:
  explicit SimpleInstance(PP_Instance instance) : pp::Instance(instance) {
  }

  virtual void HandleMessage(const pp::Var& var_message) {
    if (var_message.is_string() && var_message.AsString() == "ping") {
      PostMessage(pp::Var("pong"));
      return;
    }
    PostMessage(pp::Var("failed"));
  }
};

class SimpleModule : public pp::Module {
 public:
  virtual pp::Instance* CreateInstance(PP_Instance instance) {
    return new SimpleInstance(instance);
  }
};

namespace pp {

__attribute__((visibility("default")))
Module* CreateModule() {
  return new SimpleModule();
}

}  // namespace pp

File: chromeos/ash/components/audio/device_selection_test_gen/tests/simple.py
# Copyright 2025 The Chromium Authors
# Use of this source code is governed by a BSD-style license that can be
# found in the LICENSE file.


from dsp import T, Simulator


def run(s: Simulator):
    A, B, C = s.abc(T.USB, T.USB, T.USB)

    s.plug(A)

    s.plug(B)
    s.select(B)

    s.plug(C)
    s.select(A)

    s.unplug(C)
